From bd023294002e809e7655b2e48c904dd31ef73492 Mon Sep 17 00:00:00 2001
From: sakuramilk <c.sakuramilk@gmail.com>
Date: Sun, 12 Aug 2012 13:39:33 +0900
Subject: [PATCH] init: add optiomize option

---
 init/Kconfig |   19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/init/Kconfig b/init/Kconfig
index 3354a33..dbfbbc7 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -980,13 +980,30 @@ source "usr/Kconfig"
 
 endif
 
+choice
+	prompt "Build Optimize"
+	default CC_OPTIMIZE_FOR_NORMAL
+	help
+	   This option select a build optimize.
+
 config CC_OPTIMIZE_FOR_SIZE
 	bool "Optimize for size"
 	help
 	  Enabling this option will pass "-Os" instead of "-O2" to gcc
 	  resulting in a smaller kernel.
 
-	  If unsure, say Y.
+config CC_OPTIMIZE_FOR_NORMAL
+	bool "Optimize for normal"
+	help
+	  Enabling this option will pass "-O2" to gcc
+	  resulting in a better size and speed kernel.
+
+config CC_OPTIMIZE_FOR_SPEED
+	bool "Optimize for speed"
+	help
+	  Enabling this option will pass "-O3" instead of "-O2" to gcc
+	  resulting in a good speed kernel.
+endchoice
 
 config CC_CHECK_WARNING_STRICTLY
 	bool "Check warnings more strictly"
-- 
1.7.10
